SimpleClientDHTSSLWIFIMGR/network-selector.html

```diff
@@ -0,0 +1,484 @@
+  console.log("Submitting AP", ssid, password);
+}
+
+function onClickAP() {
+  console.log("Clicking", this, this.dataset);
+  var ssid = this.dataset.ssid;
+  var encrypted = this.dataset.encrypted;
+  if (encrypted) {
+    activeSsid = ssid; // store this globally for submitting later
+    var passwordScreen = $$(".passwordScreen");
+    passwordScreen.querySelector(".network-name").textContent = ssid;
+    passwordScreen.classList.add('enabled');
+    passwordScreen.querySelector(".passwordInput").focus();
+  } else {
+    submitAP(ssid);
+  }
+}
+
+function loadData(knownAPs) {
+  var container = $$('.access-points-container');
+  while (container.firstChild) {
+    container.firstChild.removeEventListener('click', onClickAP);
+    container.removeChild(container.firstChild);
+  }
+  
+  knownAPs.sort((a,b) => 
+    b.rssi - a.rssi
+  ).map(x => {
+    var rootElement = document.importNode(accessPointTpl.content, true).children[0];
+    
+    rootElement.querySelector('.access-point-ssid').textContent = x.ssid;
+    rootElement.querySelector('.access-point-encryption').classList.toggle('icon-lock', x.encryptionType.length)
+    var indicators = rootElement.querySelectorAll('.access-point-rssi-indicator');
+    indicators[2].classList.toggle('enabled', x.rssi >= 1);
+    indicators[1].classList.toggle('enabled', x.rssi >= 2);
+    indicators[0].classList.toggle('enabled', x.rssi >= 3);
+    
+    rootElement.dataset['ssid'] = x.ssid;
+    
+    if (x.encryptionType.length)
+      rootElement.dataset['encrypted'] = "true"; 
+    
+    rootElement.addEventListener('click', onClickAP);
+    
+    return rootElement;
+  }).forEach(x => container.appendChild(x) );
+  
+}
+
+window.addEventListener('error', function(e) {
+  $$(".status").textContent = "Whoops, something went wrong. Reset your weather station and try again. ";
+  $$(".status").textContent += "The nerdy details: \"" + e.message + "\"";
+  $$(".access-points-container").style.display = "none";
+});
+
+</script>
+<script>
+
+// Array of APs
+// ssid: String
+// rssi: Number, integer 0-3 incl.
+// encryptionType: String, "" | *
+var knownAPs = [
+  {
+    ssid: "SPARK-JB9NCX",
+    rssi: 0,
+    encryptionType: "WEP"
+  },
+  {
+    ssid: "Dunedin Public Wifi",
+    rssi: 1,
+    encryptionType: "" // open network
+  },
+  {
+    ssid: "Donald Trump Internet AP",
+    rssi: 2,
+    encryptionType: "" // open network
+  },
+  {
+    ssid: "Surveillance Van #5",
+    rssi: 3,
+    encryptionType: "WEP"
+  },
+  {
+    ssid: "Serious Business Network",
+    rssi: 0,
+    encryptionType: "WPA2"
+  },
+  {
+    ssid: "Surveillance Van #8",
+    rssi: 1,
+    encryptionType: "WPA2"
+  },
+  {
+    ssid: "Surveillance Van #3",
+    rssi: 2,
+    encryptionType: "WPA2"
+  },
+  {
+    ssid: "Vehicle for Agile Distribution of Dairy-based Colloidal Foams (VAD-DCF)",
+    rssi: 3,
+    encryptionType: "WPA2"
+  },
+];
+
+function refreshData() {
```

Comment:
This refresh function is just for testing at the moment, but this could made more real/useful. Or just deleted.